test/testhelper/client/k8s/client.go

```diff
@@ -0,0 +1,284 @@
+}
+
+func (c *Client) listNamespaces() (*v1.NamespaceList, error) {
+	namespaceList, err := c.client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get namespaces")
+	}
+	return namespaceList, nil
+}
+
+func (c *Client) ListNodesByLabel(label string) (*v1.NodeList, error) {
+	nodes, err := c.client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{LabelSelector: label})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get cluster nodes")
+	}
+	return nodes, nil
+}
+
+func (c *Client) ListNodeIPByLabel(label string) ([]net.IP, error) {
+	var ips []net.IP
+	nodes, err := c.ListNodesByLabel(label)
+	if err != nil {
+		return nil, err
+	}
+	for _, node := range nodes.Items {
+		for _, v := range node.Status.Addresses {
+			if v.Type == v1.NodeInternalIP {
+				ips = append(ips, net.ParseIP(v.Address))
+			}
+		}
+	}
+	return ips, nil
+}
+
+func (c *Client) ListAllNamespacesPods() ([]*NamespacePod, error) {
+	namespaceList, err := c.listNamespaces()
+	if err != nil {
+		return nil, err
+	}
+	var namespacePodList []*NamespacePod
+	for _, ns := range namespaceList.Items {
+		pods, err := c.client.CoreV1().Pods(ns.Name).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get all namespace pods")
+		}
+		namespacePod := NamespacePod{
+			Namespace: ns,
+			PodList:   pods,
+		}
+		namespacePodList = append(namespacePodList, &namespacePod)
+	}
+
+	return namespacePodList, nil
+}
+
+func (c *Client) ListKubeSystemPodsStatus() (bool, error) {
+	pods, err := c.client.CoreV1().Pods("kube-system").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to get kube-system namespace pods")
+	}
+	// pods.Items maybe nil
+	if len(pods.Items) == 0 {
+		return false, nil
+	}
+	for _, pod := range pods.Items {
+		// pod.Status.ContainerStatus == nil because of pod contain initcontainer
+		if len(pod.Status.ContainerStatuses) == 0 {
+			continue
+		}
+		if !pod.Status.ContainerStatuses[0].Ready {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (c *Client) GetPodLog(nameSpace, podName string) (string, error) {
+	req := c.client.CoreV1().Pods(nameSpace).GetLogs(podName, &v1.PodLogOptions{})
+	podLogs, err := req.Stream(context.TODO())
+	if err != nil {
+		return "", err
+	}
+
+	defer func() {
+		_ = podLogs.Close()
+	}()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	str := buf.String()
+	return str, nil
+}
+
+func (c *Client) GetPodEvents(nameSpace, podName string) ([]v1.Event, error) {
+	events, err := c.client.CoreV1().
+		Events(nameSpace).
+		List(context.TODO(), metav1.ListOptions{FieldSelector: "involvedObject.name=" + podName, TypeMeta: metav1.TypeMeta{Kind: "Pod"}})
+	if err != nil {
+		return nil, err
+	}
+	return events.Items, nil
+}
+
+func (c *Client) getPodReadyStatus(pod v1.Pod) error {
```

Comment:
I think this function only checks if the pod is ready, so returning true/false would be good. What do you think?

Reply:
Make sense